elasticdl/python/tests/test_helper.py

```diff
@@ -0,0 +1,213 @@
+                    ),
+                    "label": tf.train.Feature(
+                        int64_list=tf.train.Int64List(value=[label])
+                    ),
+                }
+            elif dataset_name == DatasetName.TEST_MODULE:
+                x = np.random.rand(shape).astype(np.float32)
+                y = 2 * x + 1
+                example_dict = {
+                    "x": tf.train.Feature(
+                        float_list=tf.train.FloatList(value=x)
+                    ),
+                    "y": tf.train.Feature(
+                        float_list=tf.train.FloatList(value=y)
+                    ),
+                }
+            elif dataset_name == DatasetName.IMAGE_DEFAULT:
+                image = np.random.rand(np.prod(shape)).astype(np.float32)
+                label = np.ndarray([1], dtype=np.int64)
+                label[0] = np.random.randint(0, 10)
+                example_dict = {
+                    "image": tf.train.Feature(
+                        float_list=tf.train.FloatList(value=image)
+                    ),
+                    "label": tf.train.Feature(
+                        int64_list=tf.train.Int64List(value=[label])
+                    ),
+                }
+            else:
+                raise ValueError("Unknown dataset name %s." % dataset_name)
+
+            example = tf.train.Example(
+                features=tf.train.Features(feature=example_dict)
+            )
+            f.write(example.SerializeToString())
+    return temp_file.name
+
+
+def distributed_train_and_evaluate(
+    feature_shape,
+    model_zoo_path,
+    model_def,
+    model_params="",
+    training=True,
+    dataset_name=DatasetName.IMAGE_DEFAULT,
+    callback_classes=[],
+    use_async=False,
+    get_model_steps=1,
+):
+    """Runs distributed training and evaluation with a local master. Grpc
+    calls are mocked by local master call.
+
+    Args:
+        feature_shape: The shape of model input.
+        model_zoo_path: The directory that contains user-defined model files
+            or a specific model file.
+        model_def: The import path to the model definition function/class in
+            the model zoo, e.g.  "cifar10_subclass.CustomModel".
+        model_params: The dictionary of model parameters in a string that will
+            be used to instantiate the model, e.g. "param1=1,param2=2".
+        training: True for job type `TRAIN_WITH_EVALUATION`, False for
+            job type `EVALUATION`.
+        dataset_name: A dataset name from `DatasetName`.
+        callback_classes: A List of callbacks that will be called at given
+            stages of the training procedure.
+        use_async: A python bool. True if using asynchronous updates.
+        get_model_steps: Worker will perform `get_model` from the parameter
+            server every this many steps.
+
+    Returns:
+        An integer indicating the version of trained model in master.
```

Comment:
model version after the distributed training and evaluation.

Reply:
Done.